### DISCUSSION TITLE
Configure multiple projects

### DISCUSSION THREAD

**Discussion by lethienhung** [invertase/react-native-firebase#6463] [Q&A]
Can I configure one app in one project as the default app and use a second app from another as a secondary app?

Currently, I've configured the second app from a different project, when I use it for authentication via the phone number, it works properly on the simulator, but totally the opposite when using an actual device to test, it throws out an error 
on iOS:
`[Error: [auth/invalid-app-credential] Invalid token.]` 
and on Android: 
`[Error: [auth/app-not-authorized] This app is not authorized to use Firebase Authentication. Please verify that the correct package name and SHA-1 are configured in the Firebase Console. [ A safety_net_token was passed, but no matching SHA-256 was registered in the Firebase console. Please make sure that this application’s packageName/SHA256 pair is registered in the Firebase Console. ]]`

**Comment by mikehardy** [comment]
Hi there! Secondary apps are supported for certain modules (https://rnfb-docs.netlify.app/app/usage#secondary-apps) and auth appears to be one of them. I'm not 100% sure of the interaction with app check though, as that does *not* appear to be one of the ones that supports multiple apps.

It may be that you cannot turn on app check enforcing on the secondary app, I've never tried it.

Does it work without app check integrated, as a test?

**Comment by lethienhung** [comment]
Hi @mikehardy! 
Thanks for your quick reply. I don't install the `@react-native-firebase/app-check` for my app. I've just followed instructions on how to set up the default app, then I initialize another app with different firebase credentials. I've added the **Auth Key p8** file to the **Firebase Console** alongside **SHA-1** and **SHA-256** for android on both apps. I've also added REVERSED_CLIENT_ID, APP_ID of both two apps to URL Schemes. Please see screenshots

In AppDelegate.m, I've added:
```
#import <Firebase.h>
...
[FIRApp configure];
```

In JS usage:
```
import auth, {FirebaseAuthTypes} from '@react-native-firebase/auth';
...
try {
  let app = firebase.apps.find(mApp => mApp.name === secondaryAppName);
  if(!app) {
    app = await firebase.initializeApp(credentials, config)
  }
} catch (error) {}

...

auth(secondaryApp).signInWithPhoneNumber(..)
```

![Screen Shot 2022-08-12 at 11 31 06](https://user-images.githubusercontent.com/9305217/184286226-ba962588-a915-4007-b007-d8de40aa896d.png)
![Screen Shot 2022-08-12 at 11 33 51](https://user-images.githubusercontent.com/9305217/184286232-b604ee59-a692-4b67-ab83-504d4c720125.png)
![Screen Shot 2022-08-12 at 11 37 37](https://user-images.githubusercontent.com/9305217/184286240-847ac3df-3cb8-40e6-8e07-25e8cb6d2c33.png)

**Reply by mikehardy** [reply]
it may be that you are not correctly initializing the second app, and not getting the auth instance from it, I'm not sure. It seems if it works on emulator it should work on a real device, but I've never tried this setup so I just don't know, sorry.

It's possible for you to add logging statements all through our stack (javascript, and our java) to verify API calls and the app they are working with, and their arguments and return values) to inspect further though, the source is open so people may conduct deeper investigations as needed
